packages/components/src/navigator/navigator-provider/component.tsx

```diff
@@ -30,26 +23,177 @@ import type {
 	NavigatorProviderProps,
 	NavigatorLocation,
 	NavigatorContext as NavigatorContextType,
+	NavigateOptions,
 	Screen,
+	NavigateToParentOptions,
 } from '../types';
 
 type MatchedPath = ReturnType< typeof patternMatch >;
-type ScreenAction = { type: string; screen: Screen };
+type RouterAction =
+	| { type: 'add' | 'remove'; screen: Screen }
+	| { type: 'goback' }
+	| { type: 'goto'; path: string; options?: NavigateOptions }
+	| { type: 'gotoparent'; options?: NavigateToParentOptions };
 
 const MAX_HISTORY_LENGTH = 50;
```

Comment:
I couldn't help but wonder what's the rationale behind keeping such a short history, and whether it would be costly to increase it. Could potentially turn useful as the router gets used more and more. Not something we should aim for in this PR of course.

Reply:
This is probably not needed. There are `createMemoryHistory` functions both in `history` and in `@tanstack/react-router`, and neither does bother to limit the history size. You can freely `push` new items to the array with no limit.

I'll do a followup that removes this. I also suspect that the `replace` action doesn't work very well.